libs/partners/groq/langchain_groq/chat_models.py

```diff
@@ -85,15 +85,222 @@ class ChatGroq(BaseChatModel):
     To use, you should have the
     environment variable ``GROQ_API_KEY`` set with your API key.
 
-    Any parameters that are valid to be passed to the groq.create call can be passed
-    in, even if not explicitly saved on this class.
+    Any parameters that are valid to be passed to the groq.create call 
+    can be passed in, even if not explicitly saved on this class.
 
     Example:
         .. code-block:: python
 
             from langchain_groq import ChatGroq
 
             model = ChatGroq(model_name="mixtral-8x7b-32768")
+
+    Setup:
+        Install ``langchain-groq`` and set environment variable 
+        ``GROQ_API_KEY``.
+
+        .. code-block:: bash
+
+            pip install -U langchain-groq
+            export GROQ_API_KEY="your-api-key"
+
+    Key init args — completion params:
+        model_name: str
+            Name of Groq model to use. E.g. "mixtral-8x7b-32768".
+        temperature: float
+            Sampling temperature. Ranges from 0.0 to 1.0.
+        max_tokens: Optional[int]
+            Max number of tokens to generate.
+
+    Key init args — client params:
+        cache: Union[BaseCache, bool, None]
+            Whether to cache the response. Uses global cache if true, does not
+            use a cache if false, uses global cache (if it is set) if None,
+            and uses the provided cache if is instance of BaseCache.
+        callbacks: Callbacks
+            Callbacks to add to the run trace.
```

Comment:
i think we probably just want to stick to the groq client params here, don't need to document cache and callbacks. model cache is usually better specified globally https://python.langchain.com/v0.2/docs/how_to/chat_model_caching/ and callbacks are generally better passed in at runtime